fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/WorldMixin.java

```diff
@@ -77,8 +79,15 @@ protected boolean onPurgeRemovedBlockEntities(List<BlockEntity> blockEntityList,
 			}
 		}
 
-		// Mimic vanilla logic
-		return blockEntityList.removeAll(removals);
+		// Mimic vanilla logic - with some perf boosts
+		// Gate perf boost behind a minimum amount of removals
+		if (removals.size() <= 4) {
```

Comment:
I suggest moving this to a static final field, like the fence used for converting a linked list to a red-black tree in hash map